src/Nethereum.JsonRpc.Client/RpcRequestResponseBatch.cs

```diff
@@ -6,31 +6,33 @@ namespace Nethereum.JsonRpc.Client
 {
     public class RpcRequestResponseBatch
     {
+        public bool AcceptPartiallySuccessful { get; set; } = false;
+
         public List<IRpcRequestResponseBatchItem> BatchItems { get; set; } = new List<IRpcRequestResponseBatchItem>();
 
         public RpcRequestMessage[] GetRpcRequests()
         {
             return BatchItems.Select(x => x.RpcRequestMessage).ToArray();
         }
 
-        public void UpdateBatchItemResponses(IEnumerable<RpcResponseMessage> responses)
+        public virtual void UpdateBatchItemResponses(IEnumerable<RpcResponseMessage> responses)
         {
             var errors = new List<RpcError>();
             foreach (var response in responses)
             {
                 var batchItem = BatchItems.First(x => x.RpcRequestMessage.Id.ToString() == response.Id.ToString());
                 if (response.HasError)
                 {
-                    errors.Add(new RpcError(response.Error.Code, response.Error.Message + ": " + batchItem.RpcRequestMessage.Method,
-                     response.Error.Data));
+                    batchItem.DecodeResponse(response);
+                    errors.Add(new RpcError(response.Error.Code, response.Error.Message + ": " + batchItem.RpcRequestMessage.Method, response.Error.Data));
```

Comment:
I could use batchItem.RpcError to add to errors, but the RpcError object wasn't created the same way, so just in case people expected a specific string already, i did not want to break their compability.